### PR TITLE
Re unite orphaned rfi files

### PR DESCRIPTION
**Pull request by rafasdc**
Implements # 1907

### CODE REVIEW

File: app/lib/theme/components/FileComponent.tsx [hunk 1]
```diff
@@ -209,23 +238,79 @@ const FileComponent: React.FC<FileComponentProps> = ({
...
+            >
+              <StyledDatePicker
+                id={id}
+                sx={getStyles(false)}
+                disabled={false}
+                readOnly={false}
+                onChange={(d: Date | null) => {
+                  if (d) {
+                    const originalDate = new Date(d);
+                    const realDate = new Date(originalDate.toDateString());
```
Comment (AntBush) [inline app/lib/theme/components/FileComponent.tsx:265]: Is this to set it without Hours/Minutes/Seconds? Why don't we just do something like `originalDate.setHours(0,0,0,0)` to zero-out everything. Would read more clear to me

File: app/lib/theme/widgets/FileWidget.tsx [hunk 2]
```diff
@@ -120,6 +123,7 @@ const FileWidget: React.FC<FileWidgetProps> = ({
           name,
           size,
           type,
+          ...(useFileDate ? { fileDate } : {}),
```
Comment (AntBush) [inline app/lib/theme/widgets/FileWidget.tsx:126]: I thought we were going to pass back the `created_at` if we weren't using the `useFileDate` here? Or is that just in a different ticket

File: app/lib/theme/components/FileComponent.tsx [hunk 3]
```diff
@@ -209,23 +238,79 @@ const FileComponent: React.FC<FileComponentProps> = ({
...
+                enUS.components.MuiLocalizationProvider.defaultProps.localeText
+              }
+              dateAdapter={AdapterDayjs}
+            >
+              <StyledDatePicker
+                id={id}
+                sx={getStyles(false)}
+                disabled={false}
+                readOnly={false}
+                onChange={(d: Date | null) => {
```
Comment (AntBush) [inline app/lib/theme/components/FileComponent.tsx:262]: When I choose a date on the date picker, I get a an error of `TypeError: value.isBefore is not a function`. I remember this is from not using dayjs when setting the date I think was the problem.

**Review by AntBush — COMMENTED** [review]
Looks good, running into a bug when setting the date

File: app/lib/theme/components/FileComponent.tsx [hunk 4]
```diff
@@ -209,23 +238,79 @@ const FileComponent: React.FC<FileComponentProps> = ({
...
-        >
-          {loading ? (
-            <LoadingSpinner color={isSecondary ? '#000000' : '#fff'} />
-          ) : (
-            buttonLabel()
-          )}
-        </StyledButton>
+      <div
+        style={
+          useFileDate
```
[the remark this answers is not in the document]
Comment (rafasdc) [reply app/lib/theme/components/FileComponent.tsx:243]: Agreed, changed

File: app/lib/theme/widgets/FileWidget.tsx (hunk 2 above)
Comment (rafasdc) [reply app/lib/theme/widgets/FileWidget.tsx:128]: That's a different ticket, it should make more sense now that the other ticket is merged. I'm using `fileDate` to capture the "Date file received" field which might be different than the uploaded date, if that makes sense.

File: app/lib/theme/components/FileComponent.tsx (hunk 1 above)
Comment (rafasdc) [reply app/lib/theme/components/FileComponent.tsx:279]: I believe this is to handle the time zone stuff and set it in the correct order, it was taken directly from the DatePickerWidget file, only difference is the setter.

File: app/lib/theme/components/FileComponent.tsx [hunk 5]
```diff
@@ -144,12 +159,25 @@ const FileComponent: React.FC<FileComponentProps> = ({
...
 }) => {
   const hiddenFileInput = useRef() as MutableRefObject<HTMLInputElement>;
   const isFiles = value?.length > 0;
   const hideIfFailed = !!error && hideFailedUpload;
   const isSecondary = buttonVariant === 'secondary';
 
+  // eslint-disable-next-line react/no-unstable-nested-components
+  const ClearableIconButton = () => {
+    return (
+      <button type="button" onClick={() => setFileDate(undefined)}>
```
[the remark this answers is not in the document]
Comment (rafasdc) [reply app/lib/theme/components/FileComponent.tsx:195]: This is a workaround as the datepicker won't accept a component with props, based of the original DatePickerWidget.

File: app/lib/theme/components/FileComponent.tsx (hunk 3 above)
Comment (rafasdc) [reply app/lib/theme/components/FileComponent.tsx:282]: This should be good, give it another try when you get a chance :)

**Pull request merged**